lib/haml/parser.rb

```diff
@@ -737,8 +741,10 @@ def parse_new_attribute(scanner)
       end
 
       return name, [:static, content.first[1]] if content.size == 1
-      return name, [:dynamic,
-        %!"#{content.each_with_object('') {|(t, v), s| s << (t == :str ? inspect_obj(v)[1...-1] : "\#{#{v}}")}}"!]
```

Comment:
Please don't introduce unnecessary changes. Obviously we can enable `# frozen_string_literal: true` if we change `''` to `''.dup`.